src/main/java/core/basesyntax/model/RegistrationServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.model;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.service.RegistrationService;
+
+public class RegistrationServiceImpl implements RegistrationService {
+    static final int AGE_PERMISION = 18;
+    static final int MIN_PASSWORD_AND_LOGIN_LENGTH = 6;
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public User register(User user) {
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_AND_LOGIN_LENGTH) {
+            throw new RegistrationException("Password length less then "
+                    + MIN_PASSWORD_AND_LOGIN_LENGTH + " characters");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_PASSWORD_AND_LOGIN_LENGTH) {
+            throw new RegistrationException("Login length less then "
+                    + MIN_PASSWORD_AND_LOGIN_LENGTH + " characters");
+        }
+        if (user.getLogin().equals(storageDao.get(user.getLogin()))) {
+            throw new RegistrationException("This user exist in storage, please change your "
+                    + "login");
+        }
+        if (user.getAge() < AGE_PERMISION || user.getAge() == null) {
```

Comment:
if age would be a wrapper class, for instance, Integer or Long, comparing it to some number can cause NullPointerException. So you have to perform any check or change approach to checking the min age.